internal/obsreportconfig/obsmetrics/obsmetrics.go

```diff
@@ -7,6 +7,7 @@
 package obsmetrics // import "go.opentelemetry.io/collector/internal/obsreportconfig/obsmetrics"
 
 const (
-	NameSep = "/"
-	Scope   = "go.opentelemetry.io/collector/obsreport"
+	NameSep   = "/"
+	MetricSep = "."
```

Comment:
On the SIG call, I suggested using `_` instead of `.` until we stabilize the metrics. The reason is that AFAIK Prometheus is working on supporting dots. So once it's ready, it'll be another breaking change for all the users using Prometheus to scrape the internal metrics. Suddenly all the metrics will be converted from `otelcol_receiver_accepted_metric_points` to `otelcol.receiver.accepted.metric_points` If we adopt `_` for now, I believe we can avoid that. 

I understand that `_` is not semantically correct from OTel's perspective, but we will have to make a big migration later anyway, so it might make sense to reduce the amount of breaking changes.

I'm happy to hear other opinions. And I'm not sure about the timeline for dots support in Prometheus. Maybe it's much further away than new stable Collector metrics.

cc @open-telemetry/collector-approvers  @open-telemetry/collector-contrib-approvers

Reply:
Otherwise, once the dots are supported in Prometheus, we might need a config option to convert `.`->`_` somewhere to preserve the behavior and avoid breaking the users. It can be even a Prometheus receiver config option. In that case, we should be good.

@dashpole, maybe you have some ideas about it?

Reply:
Note that use of dots vs underscores will be negotiated using headers, so users will be able to control it server-side: https://github.com/prometheus/proposals/blob/main/proposals/2023-08-21-utf8.md#query-time.  If it is enabled by default at some point in the future, it might still be a surprise for users.

Reply:
@dmitryax the reason i went with `.` instead of `_` is that other metrics (for example the grpc instrumentation) uses `.` already when emitting OTLP. Having both separators, depending on the source of the metric, seemed confusing.

Reply:
> Note that use of dots vs underscores will be negotiated using headers, so users will be able to control it server-side: https://github.com/prometheus/proposals/blob/main/proposals/2023-08-21-utf8.md#query-time. If it is enabled by default at some point in the future, it might still be a surprise for users.

Thanks for the clarification. Looks like, even if it is enabled by default, it won't happen any time soon. I assume the controlling headers will be exposed as a configuration in the Go OTel exporter as well as somewhere in service::telemetry::metrics` section of the collector config.

Reply:
> @dmitryax the reason i went with . instead of _ is that other metrics (for example the grpc instrumentation) uses . already when emitting OTLP. Having both separators, depending on the source of the metric, seemed confusing.

Makes sense